### PR TITLE
docs(python): Fix bullet points not rendering correctly in `DataFrame.join` docstring

### PR DESCRIPTION
**Pull request by MarcoGorelli**
once again, fixing up docstring bullet points which don't render properly 😄 check the "coalesce" arg description in https://docs.pola.rs/py-polars/html/reference/dataframe/api/polars.DataFrame.join.html

bonus drive-bys:
- some ergonomic syntax
- Series.cum_count was left out of the docs

### CODE REVIEW

File: docs/src/python/user-guide/expressions/window.py [hunk 1]
```diff
@@ -34,7 +34,7 @@
 
 # --8<-- [start:sort]
 out = filtered.with_columns(
-    pl.col(["Name", "Speed"]).sort_by("Speed", descending=True).over("Type 1"),
+    pl.col("Name", "Speed").sort_by("Speed", descending=True).over("Type 1"),
```
Comment (MarcoGorelli) [inline docs/src/python/user-guide/expressions/window.py:37]: drive-by ergonomic syntax

File: py-polars/polars/dataframe/frame.py [hunk 2]
```diff
@@ -9447,7 +9448,7 @@ def n_unique(self, subset: str | Expr | Sequence[str | Expr] | None = None) -> i
         This method operates at the `DataFrame` level; to operate on subsets at the
         expression level you can make use of struct-packing instead, for example:
 
-        >>> expr_unique_subset = pl.struct(["a", "b"]).n_unique()
+        >>> expr_unique_subset = pl.struct("a", "b").n_unique()
```
Comment (MarcoGorelli) [inline py-polars/polars/dataframe/frame.py:9451]: more drive-by

File: py-polars/polars/dataframe/frame.py [hunk 3]
```diff
@@ -9458,7 +9459,7 @@ def n_unique(self, subset: str | Expr | Sequence[str | Expr] | None = None) -> i
         In aggregate context there is also an equivalent method for returning the
         unique values per-group:
 
-        >>> df_agg_nunique = df.group_by(["a"]).n_unique()
+        >>> df_agg_nunique = df.group_by("a").n_unique()
```
Comment (MarcoGorelli) [inline py-polars/polars/dataframe/frame.py:9462]: same

File: py-polars/polars/lazyframe/frame.py [hunk 4]
```diff
@@ -3936,6 +3936,7 @@ def join(
             Join on null values. By default null values will never produce matches.
         coalesce
             Coalescing behavior (merging of join columns).
+
```
Comment (MarcoGorelli) [inline py-polars/polars/lazyframe/frame.py:3939]: this empty line is the gist of the pr

**Review by stinodego — APPROVED** [review]
Thanks :ok_hand:

**Comment by alexander-beedie** [comment]
While you're in those methods I spotted another one 😁 

```
This is currently not supported the streaming engine.
```
```
This is currently not supported by the streaming engine.
```
